loader/src/main/java/net/neoforged/neoforgespi/locating/ModFileDiscoveryAttributes.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright (c) NeoForged and contributors
+ * SPDX-License-Identifier: LGPL-2.1-only
+ */
+
+package net.neoforged.neoforgespi.locating;
+
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Attributes of a modfile relating to how it was discovered.
+ *
+ * @param parent        The mod file that logically contains this mod file.
+ * @param reader        The reader that was used to get a mod-file from jar contents. May be null if the mod file was directly created by a locator.
+ * @param systemModFile True if this mod file is supplied by the modding system and is searched for basic system mods.
+ */
+public record ModFileDiscoveryAttributes(@Nullable IModFile parent,
+        @Nullable IModFileReader reader,
+        @Nullable IModFileCandidateLocator locator,
+        @Nullable IDependencyLocator dependencyLocator,
+        boolean systemModFile) {
+
+    public static final ModFileDiscoveryAttributes DEFAULT = new ModFileDiscoveryAttributes(null, null, null, null, false);
+    public ModFileDiscoveryAttributes withSystemModFile(boolean systemModFile) {
```

Comment:
I don't think that NeoForge will be correctly marked as a system mod when it is loaded via a plain locator?

I think that we can remove the whole declaration of system mods and hard code `minecraft` and `neoforge` instead.